dotCMS/src/integration-test/java/com/dotcms/content/elasticsearch/util/ESMappingUtilHelperTest.java

```diff
@@ -0,0 +1,570 @@
+
+            //Create a working index
+            boolean result = contentletIndexAPI.createContentIndex(workingIndex);
+            //Validate
+            assertTrue(result);
+
+            //verify mapping
+            final String mapping = esMappingAPI.getMapping(workingIndex);
+
+            //parse json mapping and validate
+            assertNotNull(mapping);
+
+            final JSONObject propertiesJSON = (JSONObject) (new JSONObject(mapping)).get("properties");
+            final JSONObject contentTypeJSON = (JSONObject) ((JSONObject) propertiesJSON
+                    .get(parentContentType.variable().toLowerCase())).get("properties");
+
+            //validate age mapping results
+            final Map ageMapping = ((JSONObject) contentTypeJSON
+                    .get(ageField.variable().toLowerCase())).getAsMap();
+            assertNotNull(ageMapping);
+            assertEquals(1, ageMapping.size());
+            assertEquals("long", ageMapping.get("type"));
+
+            //validate relationship field mapping results
+            final Map relationshipMapping = ((JSONObject) contentTypeJSON
+                    .get(relationshipField.variable().toLowerCase())).getAsMap();
+            assertNotNull(relationshipMapping);
+            assertEquals(1, relationshipMapping.size());
+            assertEquals("text", relationshipMapping.get("type"));
+
+            //validate legacy relationship
+            final Map legacyRelationshipMapping = ((JSONObject) propertiesJSON
+                    .get(legacyRelationship.getRelationTypeValue().toLowerCase())).getAsMap();
+            assertNotNull(ageMapping);
+            assertEquals(2, legacyRelationshipMapping.size());
+            assertEquals("keyword", legacyRelationshipMapping.get("type"));
+            assertEquals(8191, legacyRelationshipMapping.get("ignore_above"));
+
+        } finally {
+            if (workingIndex != null) {
+                contentletIndexAPI.delete(workingIndex);
+            }
+
+            if (parentContentType != null && parentContentType.inode() != null) {
+                ContentTypeDataGen.remove(parentContentType);
+            }
+
+            if (childContentType != null && childContentType.inode() != null) {
+                ContentTypeDataGen.remove(childContentType);
+            }
+        }
+    }
+
+    /**
+     * <b>Test Case:</b> Validates dynamic mappings defined in the `es-content-mapping` file are applied correctly<p></p>
+     * <b>Expected Results:</b> Each test case should match the `expectedResult` value, which is a string with the correct datatype in ES
+     * @param testCase
+     * @param fields
+     * @param expectedResult
+     * @throws DotDataException
+     * @throws IOException
+     * @throws DotSecurityException
+     */
+    @UseDataProvider
+    @Test
+    public void testValidateNewsLikeMapping(final String testCase, final String[] fields, final String expectedResult)
```

Comment:
Codacy found an issue: [JUnit tests should include assert() or fail()](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5739222)